From 08ed3df9bb2414f92daaed722a4f317d02416357 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kry=C5=A1tof=20=C4=8Cern=C3=BD?=
 <krystof.cerny@unipi.technology>
Date: Mon, 16 Feb 2026 12:06:17 +0100
Subject: [PATCH] net: phy: realtek: Add support for PHY LEDs on RTL8211F-VD
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Realtek RTL8211F-VD has the same LED configuration as RTL8211F.
Adding the support for `netdev` trigger.

Signed-off-by: Kryštof Černý <krystof.cerny@unipi.technology>
---
 drivers/net/phy/realtek.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/phy/realtek.c b/drivers/net/phy/realtek.c
index 8ce5705af..4e4e121ed 100644
--- a/drivers/net/phy/realtek.c
+++ b/drivers/net/phy/realtek.c
@@ -1362,6 +1362,9 @@ static struct phy_driver realtek_drvs[] = {
 		.read_page	= rtl821x_read_page,
 		.write_page	= rtl821x_write_page,
 		.flags		= PHY_ALWAYS_CALL_SUSPEND,
+		.led_hw_is_supported = rtl8211f_led_hw_is_supported,
+		.led_hw_control_get = rtl8211f_led_hw_control_get,
+		.led_hw_control_set = rtl8211f_led_hw_control_set,
 	}, {
 		.name		= "Generic FE-GE Realtek PHY",
 		.match_phy_device = rtlgen_match_phy_device,
-- 
2.47.3

